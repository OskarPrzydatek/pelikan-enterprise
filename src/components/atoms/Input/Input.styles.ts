import styled from 'styled-components';

export const Input = styled.input`
  appearance: none;
  border: none;
  width: 100%;
  line-height: 1.6;
  padding: 0.5%;
  font-family: ${({ theme }) => theme.fonts.main};
  color: ${({ theme }) => theme.colors.black};
  border: 3px solid ${({ theme }) => theme.colors.darkBlue};
  font-size: ${({ theme }) => theme.fontSizes.xs};

  @media ${({ theme }) => theme.breakpoints.phone} {
    font-size: ${({ theme }) => theme.fontSizes.xs};
  }

  @media ${({ theme }) => theme.breakpoints.tablet} {
    font-size: ${({ theme }) => theme.fontSizes.s};
  }

  @media ${({ theme }) => theme.breakpoints.desktop} {
    font-size: ${({ theme }) => theme.fontSizes.m};
  }

  &:focus {
    outline: none;
    border: 3px solid ${({ theme }) => theme.colors.lightBlue};
  }
`;
