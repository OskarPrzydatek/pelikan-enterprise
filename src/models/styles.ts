import {
  DefaultTheme,
  FlattenInterpolation,
  ThemeProps,
} from 'styled-components';

export interface ICSS {
  css?: FlattenInterpolation<ThemeProps<DefaultTheme>>;
}
