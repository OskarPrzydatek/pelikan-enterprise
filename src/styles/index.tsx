export * from './layouts';
export { theme } from './theme';
