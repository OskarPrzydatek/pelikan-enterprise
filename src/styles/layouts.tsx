import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyleLayout = createGlobalStyle`
  * {
		box-sizing: border-box;
    outline: none;
	} 

  body {
    margin: 0;
    padding: 0;
  }

  #root {
    height: 100vh;
    display: grid;
    grid-template-rows: .5fr 8fr .5fr;
  }

	a {
		color: inherit;
		text-decoration: none;
	}

	ul {
		padding: 0;
    margin: 0;
		list-style-type: none;
	}
`;

export const MainLayout = styled.main`
  padding: ${({ theme }) => theme.spaces.l};
`;

export const PageLayout = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
