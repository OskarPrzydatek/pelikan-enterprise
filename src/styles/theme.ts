import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  colors: {
    red: '#F23557',
    lightBlue: '#98BFF2',
    darkBlue: '#013059',
    gray: '#5F5F5F',
    black: '#2F2F2F',
    opacityBlack: '#2F2F2FCC',
    shadowBlack: '#00000040',
    white: '#FFFFFF',
  },
  breakpoints: {
    phone: '(min-width: 420px)',
    tablet: '(min-width: 920px)',
    desktop: '(min-width: 1200px)',
  },
  fonts: {
    main: `'Inter', sans-serif`,
  },
  fontSizes: {
    xxs: '.2rem',
    xs: '.5rem',
    s: '1rem',
    m: '1.2rem',
    l: '1.5rem',
    xl: '1.8rem',
    xxl: '2rem',
  },
  spaces: {
    xxs: '.2rem',
    xs: '.5rem',
    s: '.8rem',
    m: '1.2rem',
    l: '1.5rem',
    xl: '1.8rem',
    xxl: '2rem',
  },
};
