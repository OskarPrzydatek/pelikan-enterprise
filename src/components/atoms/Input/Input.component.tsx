import * as S from './Input.styles';

interface IInput {
  id: string;
  placeholder: string;
  dataTestID?: string;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  onFocus?: () => void;
  onBlur?: () => void;
}

export const Input: React.FC<IInput> = ({
  id,
  placeholder,
  dataTestID,
  onChange,
  onFocus,
  onBlur,
}: IInput) => {
  return (
    <S.Input
      data-testid={dataTestID}
      id={id}
      placeholder={placeholder}
      type="text"
      onBlur={onBlur}
      onChange={onChange}
      onFocus={onFocus}
    />
  );
};
