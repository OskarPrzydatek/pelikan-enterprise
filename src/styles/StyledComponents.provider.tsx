import { ThemeProvider } from 'styled-components';

import { GlobalStyleLayout } from './layouts';
import { theme } from './theme';

interface IStyledComponentsProvider {
  children: React.ReactNode;
}

export const StyledComponentsProvider: React.FC<IStyledComponentsProvider> = ({
  children,
}: IStyledComponentsProvider) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyleLayout />
      {children}
    </ThemeProvider>
  );
};
