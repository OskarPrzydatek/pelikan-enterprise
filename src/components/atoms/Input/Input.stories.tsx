import { StyledComponentsProvider } from '~/styles';

import { Input } from './Input.component';

import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Input> = {
  title: 'Atoms/Input',
  component: Input,
};

export default meta;
type Story = StoryObj<typeof Input>;

export const TextInput: Story = {
  render: () => (
    <StyledComponentsProvider>
      <Input
        id="text-input"
        placeholder="Text Input"
        onBlur={() => console.log('onBlur')}
        onFocus={() => console.log('onFocus')}
        onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
          console.log(event.target.value)
        }
      />
    </StyledComponentsProvider>
  ),
};
