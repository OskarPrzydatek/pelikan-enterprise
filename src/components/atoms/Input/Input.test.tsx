import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, test, vi } from 'vitest';

import { StyledComponentsProvider, theme } from '~/styles';

import { Input } from './Input.component';

const mockOnChange = vi.fn();
const mockOnFocus = vi.fn();
const mockOnBlur = vi.fn();

describe('Input', () => {
  test('component snapshot', () => {
    const view = render(
      <StyledComponentsProvider>
        <Input
          dataTestID="atom-input"
          id="atom-input"
          placeholder="Input atom component"
          onBlur={mockOnBlur}
          onChange={mockOnChange}
          onFocus={mockOnFocus}
        />
      </StyledComponentsProvider>
    );
    expect(view).toMatchSnapshot();
  });

  test('ensure input is wtitable', () => {
    render(
      <StyledComponentsProvider>
        <Input
          dataTestID="atom-input"
          id="atom-input"
          placeholder="Input Atom Component"
          onChange={mockOnChange}
        />
      </StyledComponentsProvider>
    );
    const input: HTMLInputElement = screen.getByTestId('atom-input');
    fireEvent.change(input, { target: { value: 'abcd' } });
    expect(input.value).toBe('abcd');
    expect(mockOnChange).toHaveBeenCalled();
  });

  test('ensure input focus and blur works correctly', () => {
    render(
      <StyledComponentsProvider>
        <Input
          dataTestID="atom-input"
          id="atom-input"
          placeholder="Input Atom Component"
          onBlur={mockOnBlur}
          onFocus={mockOnFocus}
        />
      </StyledComponentsProvider>
    );
    const input: HTMLInputElement = screen.getByTestId('atom-input');
    expect(input).toHaveStyle(`border: 3px solid ${theme.colors.darkBlue}`);
    input.focus();
    expect(input).toHaveStyle(`border: 3px solid ${theme.colors.lightBlue}`);
    expect(mockOnFocus).toHaveBeenCalled();
    input.blur();
    expect(input).toHaveStyle(`border: 3px solid ${theme.colors.darkBlue}`);
    expect(mockOnBlur).toHaveBeenCalled();
  });
});
