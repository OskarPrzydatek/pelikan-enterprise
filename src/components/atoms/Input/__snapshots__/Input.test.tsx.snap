// Vitest Snapshot v1

exports[`Input > component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  width: 100%;
  line-height: 1.6;
  padding: 0.5%;
  font-family: 'Nanum Gothic',sans-serif;
  color: #21313F;
  border: 3px solid #013059;
  font-size: .8rem;
}

.c0:focus {
  outline: none;
  border: 3px solid #98BFF2;
}

@media (min-width:390px) {
  .c0 {
    font-size: .8rem;
  }
}

@media (min-width:834px) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:1728px) {
  .c0 {
    font-size: 1.4rem;
  }
}

<body>
    <div>
      <input
        class="c0"
        data-testid="atom-input"
        id="atom-input"
        placeholder="Input atom component"
        type="text"
      />
    </div>
  </body>,
  "container": .c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  width: 100%;
  line-height: 1.6;
  padding: 0.5%;
  font-family: 'Nanum Gothic',sans-serif;
  color: #21313F;
  border: 3px solid #013059;
  font-size: .8rem;
}

.c0:focus {
  outline: none;
  border: 3px solid #98BFF2;
}

@media (min-width:390px) {
  .c0 {
    font-size: .8rem;
  }
}

@media (min-width:834px) {
  .c0 {
    font-size: 1rem;
  }
}

@media (min-width:1728px) {
  .c0 {
    font-size: 1.4rem;
  }
}

<div>
    <input
      class="c0"
      data-testid="atom-input"
      id="atom-input"
      placeholder="Input atom component"
      type="text"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
